qt/python/mantidqt/widgets/memorywidget/memoryview.py

```diff
@@ -71,7 +71,14 @@ def set_bar_color(self, current_value: int, new_value: int):
         else:
             pass
 
-    def set_value(self, new_value: int, mem_used: float, mem_avail: float):
+    def invoke_set_value(self, new_value: int, mem_used: float, mem_avail: float):
+        new_value = Q_ARG(int, new_value)
+        mem_used = Q_ARG(float, mem_used)
+        mem_avail = Q_ARG(float, mem_avail)
+        QMetaObject.invokeMethod(self, "_set_value", Qt.AutoConnection, new_value, mem_used, mem_avail)
```

Comment:
Could we have a `pyqtsignal`, and call emit with the params instead?
This way we can write a unit test with the signal mocked to make sure the signal actually gets called without having to involve Qt (making it testable) unlike invoking via the global QMetaObject

Reply:
I like David's simpler idea. We don't need to capture exceptions or return values so the `QAppThreadCall` is over kill. A simple private signal/slot completely inside the view should do this?